services/alerts/src/useAlert.ts

```diff
@@ -0,0 +1,54 @@
+import { useCallback, useContext, useEffect, useState } from 'react'
+import { AlertsManagerContext } from './AlertsManagerContext'
+import { Alert, AlertOptions } from './types'
+
+const defaultOptions: AlertOptions = {
+    duration: 3000,
+}
+
+export const useAlert = (
+    message: string | ((props: any) => string),
+    options?: AlertOptions | ((props: any) => AlertOptions)
+) => {
+    const alertsManager = useContext(AlertsManagerContext)
+    const [shownAlerts] = useState(new Map<Alert, number>())
+
+    const show = useCallback(
+        props => {
+            const resolvedMessage =
+                typeof message === 'function' ? message(props) : String(message)
+            const resolvedOptions = {
+                ...defaultOptions,
+                ...(typeof options === 'function' ? options(props) : options),
+            }
+            const alert = {
+                message: resolvedMessage,
+                options: resolvedOptions,
+            }
+            const timeout = setTimeout(() => {
+                alertsManager.hide(alert)
+                shownAlerts.delete(alert)
+            }, resolvedOptions.duration)
```

Comment:
We should do this differently. I think each alert should be in control of when it is ready to be removed from the alerts collection.